### ISSUE TITLE
SNETImageUpload component and CORS

### ISSUE THREAD

**Issue by ferrouswheel** [singnet/snet-betav1-dapp#188]
I wanted to allow people to upload or use a url to send an image to my services. The SNETImageUpload is great, but I've noticed that I get a CORS error if I try to use a URL.

I have disabled URL upload with `disableUrlTab={true} `, but this issue suggests that using a random URL might not be possible unless the user has access to the server hosting it. For that reason it may be worth disabling the url tab by default.

**Comment by ferrouswheel** [comment]
Did you have any thoughts on this @ramongduraes ?

**Comment by ramongduraes** [comment]
hey @ferrouswheel 
have you tried to use `allowURL={true}` ? It is supposed to allow the component to send the image URL to the services instead of the base64/byte-encoded data for both the "URL" and "Gallery" tabs. It will still render the image on screen so the user won't know the difference.

**Comment by ferrouswheel** [comment]
This issue happens when it tries to render the image on the screen.

I was expecting the image to be rendered in the UI after entering the URL.

(my services take byte-encoded data, and would fail if sent a url instead)

**Comment by ramongduraes** [comment]
I see... I've spent a lot of time trying to figure out a way to get past CORS errors but haven't really found any effective solution. 

It happens every time the component tries to get the data from `<img>` tags if the image server blocks cross-origin access (i.e.: gettyimages). Since it can still render the image on screen, the only workaround I could think of was to allow the component to send the image URL to the services, that way we can download the file in the back-end... 

Do you have any ideas?

**Comment by ramongduraes** [comment]
I wondered why I never came across that error before but thinking about it we've realized that most of the websites only allow the user to upload images from their computer (Facebook, Instagram, etc.)... That way if the user really wants it, they can still download the image and send it to the component.

Btw, I'm not sure I should've added the `allowURL` parameter due to possible copyright infringements (we're processing images that are owned by someone else)...

**Comment by ferrouswheel** [comment]
I'm not sure if it's possible, but have you tried setting the src attribute in an image to the url and then reading the image data after the browser renders it?

**Comment by ferrouswheel** [comment]
While I'm thinking about it, when our marketplace is ssl enabled it may also make it impossible to load image urls with an http origin. Perhaps we should move all image services to processing image data directly instead of loading urls?

**Comment by ramongduraes** [comment]
Yeah, that's what the component tries to do by default! That's when we get CORS errors. Which doesn't make sense in my head since in practice we already have the data... I'll look into it again!

**Comment by ferrouswheel** [comment]
Don't worry about it too much. It may well be impossible by design, and I just wanted to report it in case other people run into the same trouble. My UIs are working nicely with the upload mechanism :-)
